cmd/guacone/cmd/graphql_server.go

```diff
@@ -0,0 +1,169 @@
+	Args:  cobra.MinimumNArgs(0),
+	Run: func(cmd *cobra.Command, args []string) {
+		ctx := logging.WithLogger(context.Background())
+		logger := logging.FromContext(ctx)
+
+		opts, err := validateGraphqlServerFlags(
+			viper.GetString("gdbuser"),
+			viper.GetString("gdbpass"),
+			viper.GetString("gdbaddr"),
+			viper.GetString("realm"),
+			viper.GetString("gql-backend"),
+			viper.GetInt("gql-port"),
+			viper.GetBool("gql-debug"),
+			args)
+		if err != nil {
+			fmt.Printf("unable to validate flags: %v\n", err)
+			_ = cmd.Help()
+			os.Exit(1)
+		}
+
+		srv, err := getGraphqlServer(opts)
+		if err != nil {
+			logger.Errorf("unable to initialize graphql server: %v", err)
+			os.Exit(1)
+		}
+		http.Handle("/query", srv)
+
+		logger.Infof("graphql server running with %v backend at http://localhost:%d/query", opts.graphqlBackend, opts.graphqlPort)
+		if opts.graphqlDebug {
+			http.Handle("/", playground.Handler("GraphQL playground", "/query"))
+			logger.Infof("connect to http://localhost:%d/ for GraphQL playground", opts.graphqlPort)
+		}
+		if opts.graphqlBackend == gqlBackendTesting {
+			logger.Infof("using testing backend, test data will be bootstrapped in the API")
+		}
+
+		logger.Fatal(http.ListenAndServe(fmt.Sprintf(":%d", opts.graphqlPort), nil))
+
+	},
+}
+
+func validateGraphqlServerFlags(user string, pass string, dbAddr string, realm string,
+	graphqlBackend string, graphqlPort int, graphqlDebug bool, args []string) (graphqlServerOptions, error) {
+
+	var opts graphqlServerOptions
+	opts.user = user
+	opts.pass = pass
+	opts.dbAddr = dbAddr
+	opts.realm = realm
+
+	if graphqlBackend != gqlBackendNeo4j &&
+		graphqlBackend != gqlBackendInmem &&
+		graphqlBackend != gqlBackendTesting {
+		return opts, fmt.Errorf("invalid graphql backend specified: %v", graphqlBackend)
+	}
+
+	opts.graphqlBackend = graphqlBackend
+	opts.graphqlPort = graphqlPort
+	opts.graphqlDebug = graphqlDebug
+
+	return opts, nil
+}
+
+func getGraphqlServer(opts graphqlServerOptions) (*handler.Server, error) {
+	var topResolver resolvers.Resolver
+
+	switch opts.graphqlBackend {
+
+	case gqlBackendNeo4j:
+		args := neo4j.Neo4jConfig{
+			User:   opts.user,
+			Pass:   opts.pass,
+			Realm:  opts.realm,
+			DBAddr: opts.dbAddr,
+		}
+
+		backend, err := neo4j.GetBackend(&args)
+		if err != nil {
+			return nil, fmt.Errorf("Error creating neo4j backend: %w", err)
+		}
+
+		topResolver = resolvers.Resolver{Backend: backend}
+
+	case gqlBackendTesting:
+		args := testing.DemoCredentials{}
+		backend, err := testing.GetBackend(&args)
+		if err != nil {
+			return nil, fmt.Errorf("Error creating testing backend: %w", err)
+		}
+
+		topResolver = resolvers.Resolver{Backend: backend}
+
+	case gqlBackendInmem:
+		args := testing.DemoCredentials{}
+		backend, err := testing.GetEmptyBackend(&args)
+		if err != nil {
+			return nil, fmt.Errorf("Error creating inmem backend: %w", err)
+		}
+
+		topResolver = resolvers.Resolver{Backend: backend}
```

Comment:
I'm thinking we should merge these and have a test flag that populates the data in either inmem/neo4j backends.

Or maybe we should punt this for later when all ingestion is pushed to be done via graphql instead of directly?

Reply:
But in that case, neo4j is not populated with testing data, we'd still need a flag for that.

Reply:
yea - i feel like it wouldn't quite be needed. i'm more than happy to remove "testing" backend here. wdyt?

Reply:
Yeah, let's remove it, this way guacone is only production/real data

Reply:
removed